src/main/java/core/basesyntax/service/impl/FileReadImpl.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.FileRead;
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FileReadImpl implements FileRead {
+    @Override
+    public List<String> readDataFromFile(String fileName) {
+        List<String> lines = new ArrayList<>();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fileName))) {
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                lines.add(line);
+            }
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("File doesn't exist!");
+        } catch (IOException ioException) {
+            throw new RuntimeException("File cannot be read!");
```

Comment:
Similarly, when rethrowing the IOException, include the original exception object and provide a more informative message. This will help with debugging if an issue arises.